### PR TITLE
Remove lodash isEquals

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Part of removing lodash #991 

### WHAT is this pull request doing?

Creating a `isEquals` function and using that instead of `_.isObjectsEqual`

### How to 🎩
Test pass? 💯 

### TODO:
* add changelog

### CODE REVIEW

**Comment by AndrewMusgrave** [comment]
WIP - trying to fix a circular import issue

File: src/components/Icon/types.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import {SVGSource} from '@shopify/images';
```
Comment (chloerice) [inline src/components/Icon/types.ts]: 👌

File: src/components/Icon/types.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  | 'yellowLighter'
+  | 'yellow'
+  | 'yellowDark'
+  | 'orange'
+  | 'redLighter'
+  | 'red'
+  | 'redDark'
+  | 'purple';
+
+export const BUNDLED_ICONS = {
```
Comment (BPScott) [inline src/components/Icon/types.ts]: This isn't a type. Having it in type.ts doesn't feel right

**Comment by dleroux** [comment]
@AndrewMusgrave the original issue was getting sandboxed so I rebased this is this branch `rm-ld-isequals-rebased`. It looks like this does fix this issue: https://github.com/Shopify/polaris-react/issues/450

I didn't want to overwrite your branch because if I'm not mistaken the icon stuff for the circular import is not required anymore.

**Comment by danrosenthal** [comment]
Hey @AndrewMusgrave, is this still something you're looking to merge? Anything else you need here?

**Comment by dleroux** [comment]
We should merge it asap, it’s affecting a few partners.

**Comment by AndrewMusgrave** [comment]
As I've let this PR go a little stale. I'm going to open a new pr that'll include, addressing function comparison overhead and cyclic values

**Pull request closed without merging**